python/ray/serve/_private/api.py

```diff
@@ -244,6 +203,118 @@ def serve_start(
                 raise TimeoutError(
                     f"HTTP proxies not available after {HTTP_PROXY_TIMEOUT}s."
                 )
+    return controller, controller_name
+
+
+async def serve_start_async(
+    detached: bool = False,
+    http_options: Optional[Union[dict, HTTPOptions]] = None,
+    dedicated_cpu: bool = False,
+    **kwargs,
+) -> ServeControllerClient:
+    """Initialize a serve instance in async
+
+    This function is not thread-safe, caller should maintain the async lock in order
+    to start serve instance asynchronously.
+
+    This function has same functionality with ray.serve._private.api.serve_start().
+
+    Parameters & Returns are same as ray.serve._private.api.serve_start().
+    """
+
+    usage_lib.record_library_usage("serve")
+
+    try:
+        client = get_global_client(_health_check_controller=True)
+        logger.info(
+            f'Connecting to existing Serve app in namespace "{SERVE_NAMESPACE}".'
+            " New http options will not be applied."
+        )
+        if http_options:
+            _check_http_options(client, http_options)
+        return client
+    except RayServeException:
+        pass
+
+    controller, controller_name = await ray.remote(_start_controller).remote(
+        detached, http_options, dedicated_cpu, **kwargs
+    )
+
+    client = ServeControllerClient(
+        controller,
+        controller_name,
+        detached=detached,
+    )
+    _set_global_client(client)
+    logger.info(
+        f"Started{' detached ' if detached else ' '}Serve instance in "
+        f'namespace "{SERVE_NAMESPACE}".'
+    )
+    return client
+
+
+def serve_start(
```

Comment:
Is `serve_start()` still used anywhere? Can we remove it?

Reply:
serve_start_async will only be used by serve agent, other places will still use serve_start.